### PR TITLE
Test Suite

### PR DESCRIPTION
**Pull request by jimmoffet**

### CODE REVIEW

File: .github/workflows/deploy.yml
Comment (rahearn) [inline .github/workflows/deploy.yml:52]: [suggestion (nonblocking)] using this form was required for `api` because Christa uncovered a bug in the deployer that has now been fixed, so this whole thing can be simplified (a bit) to:

```suggestion
        push_arguments: >-
          --var DANGEROUS_SALT="$DANGEROUS_SALT"
          --var SECRET_KEY="$SECRET_KEY"
          --var AWS_ACCESS_KEY_ID="$AWS_ACCESS_KEY_ID"
          --var AWS_SECRET_ACCESS_KEY="$AWS_SECRET_ACCESS_KEY"
```

File: .github/workflows/checks.yml
Comment (rahearn) [inline .github/workflows/checks.yml:51]: [suggestion] I think we should be running `npm test` in CI as well

File: tests/app/test_config.py
Comment (rahearn) [inline tests/app/test_config.py:33]: [thought] not really related to this PR, but I'm wondering if relying on `os.environ['REDIS_URL']` is why caching is broken in prod?

File: .github/workflows/checks.yml
Comment (jimmoffet) [reply .github/workflows/checks.yml:53]: [committed](https://github.com/18F/notifications-admin/pull/20/commits/cdd7b42f21f63335c4ca3a104be935e701e19721)

File: tests/app/test_config.py
Comment (jimmoffet) [reply tests/app/test_config.py:33]: I don't believe so, I've been able to verify (by logging it from the running remote just before attempting to write to the cache) that the REDIS_URL variable is correctly set with our broker uri (the connection string itself bears another look, though). 

It happens grabs the uri value from the VCAP_SERVICES json blob at the top of config.py. This is what the UK does in their CF deployment. I suppose something about our deployment could be unique and create a sequencing issue that creates the redis client before config, but from a cursory look at app/application.py and app/__init__.py, seems like config runs at the top. Could be interesting to set the REDIS_URL manually in the env vars, so it's available at build time and see if that fixes it

File: .github/workflows/deploy.yml
Comment (jimmoffet) [reply .github/workflows/deploy.yml:52]: Do these vars need to be referenced here at all? Should we just set them directly via the cf cli instead of duplicating them in github secrets?

Comment (rahearn) [reply .github/workflows/deploy.yml:52]: I'd rather leave them here and move towards a world where no human uses the `cf` command line at all.

File: tests/app/main/views/test_templates.py
Comment (rahearn) [inline tests/app/main/views/test_templates.py:34]: Hmm, this doesn't look right to me. Did these messages somehow get the old and new versions after my PR was merged into this?

**Review by rahearn — APPROVED** [review]
Approved, pending a double check of the messages being correct rather than a merge artifact.

File: .github/workflows/deploy.yml
Comment (jimmoffet) [reply .github/workflows/deploy.yml:52]: Sounds good to me.

File: tests/app/main/views/test_templates.py
Comment (jimmoffet) [reply tests/app/main/views/test_templates.py:34]: Yeah, looks like an artifact from the most recent merge. Annoyingly, the tests still pass.

**Pull request merged**
